docs/debug/cpu_frequency.md

```diff
@@ -156,11 +156,59 @@ This can be the source of some reboot problems, but it is hard to test in a
 reliable way - reset is asynchronous event that have to take place in very short
 amount of time during transition.
 
+## Voltage regulator
+
+CPU communicates with voltage regulator using SVI2 (Serial VID Interface 2.0).
+Typically, after all frequency and voltage dependencies described above are
+resolved, transition to higher performance follows a sequence:
+
+1. SVI2 command is send to the voltage regulator, `D18F5x12C[Svi2CmdBusy]`
+   is set.
+2. Voltage regulator sets the requested voltage, waits for it to stabilize and
+   clears `Svi2CmdBusy`.
+3. After `Svi2CmdBusy` is clear frequency change is performed.
+
+Transition towards lower performance starts with frequency change and doesn't
+have to wait for it to finish before changing voltage. Because of these
+restrictions it is impossible for CPU to work with high frequency and too low
+voltage for that frequency.
+
+Multiple requests can be grouped:
+
+> * If multiple commands are issued that affect the P-state of a domain prior
+>   to when the processor initiates the change of the P-state of that domain,
+>   then the processor operates on the last one issued.
+> * Once a P-state change starts, the P-state state machine (PSSM) continues
+>   through completion unless interrupted by a PWROK deassertion. If multiple
+>   P-state changes are requested concurrently, the PSSM may group the
+>   associated VID changes separately from the associated COF changes.
```

Comment:
@krystian-hebel COF?

Reply:
Described earlier, first paragraph under the bigger table (lines 134-135)